pkg/operator/controller/nodeobservability/controller.go

```diff
@@ -157,34 +158,28 @@ func (r *NodeObservabilityReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	} else if !haveSCC {
 		return ctrl.Result{}, fmt.Errorf("failed to get securitycontextconstraints")
 	}
-	r.Log.V(1).Info("SecurityContextConstraints ensured", "Name", scc.Name)
+	r.Log.V(1).Info("securitycontextconstraint ensured", "scc.name", scc.Name)
 
 	// ensure serviceaccount
-	haveSA, sa, err := r.ensureServiceAccount(ctx, nodeObs, r.Namespace)
+	sa, err := r.ensureServiceAccount(ctx, nodeObs, r.Namespace)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("failed to ensure serviceaccount : %w", err)
-	} else if !haveSA {
-		return ctrl.Result{}, fmt.Errorf("failed to get serviceaccount")
 	}
-	r.Log.V(1).Info("ServiceAccount ensured", "Namespace", sa.Namespace, "Name", sa.Name)
+	r.Log.V(1).Info("serviceaccount ensured", "sa.namespace", sa.Namespace, "sa.name", sa.Name)
 
 	// ensure service
-	haveSvc, svc, err := r.ensureService(ctx, nodeObs, r.Namespace)
+	svc, err := r.ensureService(ctx, nodeObs, r.Namespace)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("failed to ensure service : %w", err)
-	} else if !haveSvc {
-		return ctrl.Result{}, fmt.Errorf("failed to get service")
 	}
-	r.Log.V(1).Info("Service ensured", "Namespace", svc.Namespace, "Name", svc.Name)
+	r.Log.V(1).Info("service ensured", "svc.namespace", svc.Namespace, "svc.name", svc.Name)
```

Comment:
Yeah, I had the same thought. But turns out the actual log has multiple resources with the key as `name` and thees keyvalues are appended to the log. In the end, it looks something like this (I'll put the actual logs a little later), 
```bash
{"level":"debug","ts":"2022-09-13T09:58:12.129Z","logger":"controller.nodeobservability","msg":"service ensured","reconciler group":"nodeobservability.olm.openshift.io","reconciler kind":"NodeObservability","name":"cluster","namespace":"","svc.namespace":"node-observability-operator","svc.name":"node-observability-agent"}
```
Since these are debug logs, we could print the entire resource as well. Also this log formal needs to be changed, I'm thinking of introducing a separate log package which fixes the format. (need to move from json to text format)

Reply:
Yes, true, I forgot that the logs are decorated with the request which is being reconciled. You are right let's keep this format.

> Since these are debug logs, we could print the entire resource as well. Also this log formal needs to be changed, I'm thinking of introducing a separate log package which fixes the format. (need to move from json to text format)

Well, the structured logging is what is the trend now, it's less human readable but better machine "parsable". As a matter of fact, you can switch in between the text and the JSON format by toggling `Developement` [field](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/log/zap#Options) in [the zap options](https://github.com/openshift/node-observability-operator/blob/1f92243033393e4bfdb3598531df96c0f991063e/cmd/node-observability-operator/main.go#L87-L90).